// @ali/aone-super-source-mapping-injector/esm/index.js
import { _ as _class_call_check } from "@swc/helpers/_/_class_call_check";
import { _ as _create_class } from "@swc/helpers/_/_create_class";
import { _ as _define_property } from "@swc/helpers/_/_define_property";
import { _ as _object_spread } from "@swc/helpers/_/_object_spread";
import { _ as _to_consumable_array } from "@swc/helpers/_/_to_consumable_array";
var _traverseModule_default, _generateModule_default;
// Babel依赖 - ES模块导入
import { parse } from "@babel/parser";
// @ts-ignore - Babel traverse没有类型定义
import * as traverseModule from "@babel/traverse";
// @ts-ignore - Babel generator没有类型定义  
import * as generateModule from "@babel/generator";
import * as t from "@babel/types";
// 提取实际的函数，兼容不同环境
var traverse = ((_traverseModule_default = traverseModule.default) === null || _traverseModule_default === void 0 ? void 0 : _traverseModule_default.default) || traverseModule.default || traverseModule;
var generate = ((_generateModule_default = generateModule.default) === null || _generateModule_default === void 0 ? void 0 : _generateModule_default.default) || generateModule.default || generateModule;
// 默认配置
var DEFAULT_OPTIONS = {
    enabled: true,
    getComponentName: function(fileName) {
        var _fileName_split_pop, _fileName_split;
        return (_fileName_split = fileName.split("/")) === null || _fileName_split === void 0 ? void 0 : (_fileName_split_pop = _fileName_split.pop()) === null || _fileName_split_pop === void 0 ? void 0 : _fileName_split_pop.replace(/\.[^/.]+$/, "");
    },
    includeExtensions: [
        ".jsx",
        ".tsx",
        ".js",
        ".ts"
    ],
    excludePatterns: [
        "node_modules",
        ".git",
        "dist",
        "build"
    ],
    devOnly: false
};
// 获取源码映射信息的函数
function getSourceMappingInfo(fileName, lineNumber) {
    var columnNumber = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : 0, options = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : {};
    var config = _object_spread({}, DEFAULT_OPTIONS, options);
    return {
        file: fileName,
        line: lineNumber,
        column: columnNumber,
        component: config.getComponentName(fileName)
    };
}
// 生成源码映射属性的字符串
function generateSourceMappingAttributes(sourceInfo) {
    return 'data-source-file="'.concat(sourceInfo.file, '" data-source-line="').concat(sourceInfo.line, '" data-source-component="').concat(sourceInfo.component, '"');
}
// 检查文件是否应该被处理
function shouldProcessFile(fileName) {
    var options = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
    var config = _object_spread({}, DEFAULT_OPTIONS, options);
    if (!config.enabled) {
        return false;
    }
    // 检查是否在排除模式中
    if (config.excludePatterns.some(function(pattern) {
        return fileName.includes(pattern);
    })) {
        return false;
    }
    // 检查文件扩展名
    var hasValidExtension = config.includeExtensions.some(function(ext) {
        return fileName.endsWith(ext);
    });
    if (!hasValidExtension) {
        return false;
    }
    return true;
}
// 核心转换函数
var transformCode = function(code, fileName) {
    var options = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
    if (!shouldProcessFile(fileName, options)) {
        return code;
    }
    var config = _object_spread({}, DEFAULT_OPTIONS, options);
    // 处理 JSX/TSX 文件
    if (fileName.endsWith(".jsx") || fileName.endsWith(".tsx")) {
        return transformJSXCode(code, fileName, config);
    }
    // 处理普通 JS/TS 文件
    return transformJSXCode(code, fileName, config);
};
// 转换 JSX/TSX 代码 - 使用AST解析
function transformJSXCode(code, fileName, config) {
    try {
        // 解析AST
        var ast = parse(code, {
            sourceType: "module",
            plugins: [
                "jsx",
                "typescript",
                "decorators-legacy"
            ],
            allowImportExportEverywhere: true,
            allowReturnOutsideFunction: true,
            allowSuperOutsideMethod: true,
            allowUndeclaredExports: true,
            strictMode: false,
            allowAwaitOutsideFunction: true
        });
        // 遍历AST
        traverse(ast, {
            JSXElement: function JSXElement(path) {
                try {
                    var // 添加属性到JSX元素
                    _openingElement_attributes;
                    var _openingElement_loc;
                    var node = path.node;
                    var openingElement = node.openingElement;
                    // 检查是否已经有源码映射属性
                    var hasSourceMapping = openingElement.attributes.some(function(attr) {
                        return t.isJSXAttribute(attr) && t.isJSXIdentifier(attr.name) && attr.name.name === "data-source-file";
                    });
                    if (hasSourceMapping) {
                        return;
                    }
                    // 获取文件信息
                    var lineNumber = ((_openingElement_loc = openingElement.loc) === null || _openingElement_loc === void 0 ? void 0 : _openingElement_loc.start.line) || 1;
                    var sourceInfo = getSourceMappingInfo(fileName, lineNumber, 0, config);
                    // 创建源码映射属性
                    var sourceMappingAttrs = [
                        t.jsxAttribute(t.jSXIdentifier("data-source-file"), t.stringLiteral(sourceInfo.file)),
                        t.jsxAttribute(t.jSXIdentifier("data-source-line"), t.stringLiteral(sourceInfo.line.toString())),
                        t.jsxAttribute(t.jSXIdentifier("data-source-component"), t.stringLiteral(sourceInfo.component))
                    ];
                    (_openingElement_attributes = openingElement.attributes).push.apply(_openingElement_attributes, _to_consumable_array(sourceMappingAttrs));
                } catch (elementError) {
                    console.warn("[SourceMapping] 处理JSX元素失败:", elementError.message);
                }
            }
        });
        // 生成代码 - 禁用HTML实体编码
        var result = generate(ast, {
            jsescOption: {
                minimal: true,
                quotes: "double"
            }
        }, code);
        return result.code;
    } catch (error) {
        // 如果AST解析失败，回退到原始代码
        console.warn("[SourceMapping] AST解析失败，跳过文件 ".concat(fileName, ":"), error.message);
        console.warn("[SourceMapping] 错误详情:", error);
        return code;
    }
}
// Vite插件（默认导出）
var viteAdapter = function() {
    var options = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    return {
        name: "source-mapping-injector",
        apply: "serve",
        // 确保在React插件之前执行，处理原始JSX
        enforce: "pre",
        transform: function transform(code, id) {
            return transformCode(code, id, options);
        }
    };
};
// Ice框架适配器
var iceAdapter = function() {
    var options = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    return {
        name: "ice-source-mapping-injector",
        setup: function(param) {
            var onGetConfig = param.onGetConfig;
            // 使用onGetConfig来配置构建时的 JSX 转换
            onGetConfig(function(config) {
                if (config.mode !== "production") {
                    config.transformPlugins = config.transformPlugins || [];
                    config.transformPlugins.push({
                        name: "dom-inspector-transform",
                        transform: function(code, id) {
                            return transformCode(code, id, options);
                        }
                    });
                }
            });
        }
    };
};
// 源码映射工具类
var SourceMappingInjector = /*#__PURE__*/ function() {
    "use strict";
    function SourceMappingInjector() {
        var options = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
        _class_call_check(this, SourceMappingInjector);
        _define_property(this, "options", void 0);
        this.options = _object_spread({}, DEFAULT_OPTIONS, options);
    }
    _create_class(SourceMappingInjector, [
        {
            // 转换代码
            key: "transform",
            value: function transform(code, fileName) {
                return transformCode(code, fileName, this.options);
            }
        },
        {
            // 获取源码信息
            key: "getSourceInfo",
            value: function getSourceInfo(fileName, lineNumber) {
                var columnNumber = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : 0;
                return getSourceMappingInfo(fileName, lineNumber, columnNumber, this.options);
            }
        },
        {
            // 生成属性字符串
            key: "generateAttributes",
            value: function generateAttributes(fileName, lineNumber) {
                var columnNumber = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : 0;
                var sourceInfo = this.getSourceInfo(fileName, lineNumber, columnNumber);
                return generateSourceMappingAttributes(sourceInfo);
            }
        },
        {
            // 检查文件是否应该被处理
            key: "shouldProcess",
            value: function shouldProcess(fileName) {
                return shouldProcessFile(fileName, this.options);
            }
        }
    ]);
    return SourceMappingInjector;
}();
// 浏览器环境下的全局变量暴露（仅用于工具类访问）
if (typeof window !== "undefined") {
    // 创建全局实例
    window.SourceMappingInjector = SourceMappingInjector;
}
// 命名导出所有适配器和工具
export default {
    viteAdapter: viteAdapter,
    iceAdapter: iceAdapter,
    transformCode: transformCode,
    getSourceMappingInfo: getSourceMappingInfo,
    generateSourceMappingAttributes: generateSourceMappingAttributes,
    shouldProcessFile: shouldProcessFile,
    SourceMappingInjector: SourceMappingInjector,
    DEFAULT_OPTIONS: DEFAULT_OPTIONS
};
export { viteAdapter, iceAdapter, transformCode, getSourceMappingInfo, generateSourceMappingAttributes, shouldProcessFile, SourceMappingInjector, DEFAULT_OPTIONS };
