// @ali/aone-super-dom-inspector-injector/cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}

var _class_call_check = require("@swc/helpers/_/_class_call_check");
var _create_class = require("@swc/helpers/_/_create_class");
var _define_property = require("@swc/helpers/_/_define_property");
// DOM 审查器脚本内容
var DOM_INSPECTOR_SCRIPT = "\n// DOM 审查器脚本\n(function() {\n  if (window.__domInspectorInjected) return;\n  window.__domInspectorInjected = true;\n\n  if (window.parent) {\n    window.parent.postMessage({\n      type: 'DOM_INSPECTOR_INJECTED',\n    }, '*');\n  }\n  \n  let isInspecting = false;\n  let highlightedElement = null;\n  \n  // 创建高亮样式\n  const style = document.createElement('style');\n  style.id = 'dom-inspector-style';\n  style.textContent = '.dom-inspector-highlight { outline: 3px solid #1677ff !important; outline-offset: 3px !important; background-color: rgba(22, 119, 255, 0.1) !important; } .dom-inspector-hover { outline: 3px solid #52c41a !important; outline-offset: 3px !important; cursor: crosshair !important; }';\n  document.head.appendChild(style);\n  \n  // 获取元素的 XPath\n  function getXPath(element) {\n    if (element.id) {\n      return '//*[@id=\"' + element.id + '\"]';\n    }\n    if (element === document.body) {\n      return '/html/body';\n    }\n    let ix = 0;\n    let siblings = element.parentNode.childNodes;\n    for (let i = 0; i < siblings.length; i++) {\n      let sibling = siblings[i];\n      if (sibling === element) {\n        return getXPath(element.parentNode) + '/' + element.tagName.toLowerCase() + '[' + (ix + 1) + ']';\n      }\n      if (sibling.nodeType === 1 && sibling.tagName === element.tagName) {\n        ix++;\n      }\n    }\n  }\n  \n  // 获取元素的 CSS 选择器\n  function getCSSSelector(element) {\n    if (element.id) {\n      return '#' + element.id;\n    }\n    if (element.className) {\n      const classes = element.className.split(' ').filter(c => c.trim());\n      if (classes.length > 0) {\n        return element.tagName.toLowerCase() + '.' + classes.join('.');\n      }\n    }\n    return element.tagName.toLowerCase();\n  }\n  \n  // 获取元素信息\n  function getElementInfo(element) {\n    const attributes = {};\n    for (let i = 0; i < element.attributes.length; i++) {\n      const attr = element.attributes[i];\n      attributes[attr.name] = attr.value;\n    }\n    \n    // 尝试获取源码映射信息\n    let sourceInfo = null;\n    try {      \n      // 检查是否有源码映射属性\n      if (element.hasAttribute('data-source-file') || \n          element.hasAttribute('data-source-line') || \n          element.hasAttribute('data-source-component')) {\n        sourceInfo = {\n          file: element.getAttribute('data-source-file') || '',\n          line: parseInt(element.getAttribute('data-source-line') || '0'),\n          component: element.getAttribute('data-source-component') || '',\n          column: parseInt(element.getAttribute('data-source-column') || '0')\n        };\n      }\n      \n      // 如果没有直接的源码映射，尝试从父组件获取\n      if (!sourceInfo) {\n        let parent = element.parentElement;\n        let depth = 0;\n        while (parent && depth < 5) { // 最多向上查找5层\n          if (parent.hasAttribute('data-source-file')) {\n            sourceInfo = {\n              file: parent.getAttribute('data-source-file') || '',\n              line: parseInt(parent.getAttribute('data-source-line') || '0'),\n              component: parent.getAttribute('data-source-component') || '',\n              column: parseInt(parent.getAttribute('data-source-column') || '0')\n            };\n            break;\n          }\n          parent = parent.parentElement;\n          depth++;\n        }\n      }\n      \n    } catch (error) {\n      console.warn('[DomInspector] 获取源码映射信息失败:', error);\n    }\n    \n    return {\n      tagName: element.tagName.toLowerCase(),\n      className: element.className || '',\n      id: element.id || '',\n      textContent: element.textContent?.trim().substring(0, 100) || '',\n      attributes: attributes,\n      xpath: getXPath(element),\n      cssSelector: getCSSSelector(element),\n      sourceInfo: sourceInfo\n    };\n  }\n  \n  // 高亮元素\n  function highlightElement(element) {\n    if (highlightedElement && highlightedElement.classList) {\n      highlightedElement.classList.remove('dom-inspector-highlight');\n    }\n    if (element && element.classList) {\n      element.classList.add('dom-inspector-highlight');\n      highlightedElement = element;\n    }\n  }\n  \n  // 鼠标悬停事件\n  function handleMouseOver(e) {\n    if (!isInspecting || !e.target) return;\n    e.stopPropagation();\n    if (e.target.classList) {\n      e.target.classList.add('dom-inspector-hover');\n    }\n  }\n  \n  // 鼠标离开事件\n  function handleMouseOut(e) {\n    if (!isInspecting || !e.target) return;\n    if (e.target.classList) {\n      e.target.classList.remove('dom-inspector-hover');\n    }\n  }\n  \n  // 点击选择事件\n  function handleClick(e) {\n    if (!isInspecting || !e.target) return;\n    e.preventDefault();\n    e.stopPropagation();\n    \n    const elementInfo = getElementInfo(e.target);\n    highlightElement(e.target);\n    \n    // 获取元素位置信息\n    const rect = e.target.getBoundingClientRect();\n    \n    // 发送元素信息到父窗口\n    if (window.parent) {\n      window.parent.postMessage({\n        type: 'ELEMENT_SELECTED',\n        data: elementInfo,\n        elementRect: {\n          x: rect.left,\n          y: rect.top,\n          width: rect.width,\n          height: rect.height,\n        }\n      }, '*');\n    }\n    \n    // 触发自定义事件（用于浏览器直接引入）\n    const event = new CustomEvent('dom-inspector:element-selected', {\n      detail: {\n        elementInfo: elementInfo,\n        elementRect: {\n          x: rect.left,\n          y: rect.top,\n          width: rect.width,\n          height: rect.height,\n        }\n      }\n    });\n    window.dispatchEvent(event);\n    \n    isInspecting = false;\n    if (document.body) {\n      document.body.style.cursor = '';\n    }\n    \n    // 移除事件监听器\n    document.removeEventListener('mouseover', handleMouseOver, true);\n    document.removeEventListener('mouseout', handleMouseOut, true);\n    document.removeEventListener('click', handleClick, true);\n  }\n  \n  // 开始审查模式\n  function startDOMInspection() {\n    isInspecting = true;\n    if (document.body) {\n      document.body.style.cursor = 'crosshair';\n    }\n    document.addEventListener('mouseover', handleMouseOver, true);\n    document.addEventListener('mouseout', handleMouseOut, true);\n    document.addEventListener('click', handleClick, true);\n  }\n  \n  // 停止审查模式\n  function stopDOMInspection() {\n    isInspecting = false;\n    if (document.body) {\n      document.body.style.cursor = '';\n    }\n    if (highlightedElement && highlightedElement.classList) {\n      highlightedElement.classList.remove('dom-inspector-highlight');\n      highlightedElement = null;\n    }\n    document.removeEventListener('mouseover', handleMouseOver, true);\n    document.removeEventListener('mouseout', handleMouseOut, true);\n    document.removeEventListener('click', handleClick, true);\n  }\n  \n  // 清除高亮\n  function clearHighlight() {\n    \n    // 方法1：清除当前高亮元素\n    if (highlightedElement && highlightedElement.classList) {\n      highlightedElement.classList.remove('dom-inspector-highlight');\n      highlightedElement = null;\n    }\n    \n    // 方法2：清除所有可能的高亮元素（备用方案）\n    const allHighlighted = document.querySelectorAll('.dom-inspector-highlight');\n\n    allHighlighted.forEach(element => {\n      element.classList.remove('dom-inspector-highlight');\n    });\n    \n    // 方法3：清除悬停样式\n    const allHovered = document.querySelectorAll('.dom-inspector-hover');\n\n    allHovered.forEach(element => {\n      element.classList.remove('dom-inspector-hover');\n    });\n  }\n  \n  // 监听来自父窗口的消息\n  window.addEventListener('message', function(event) {\n    if (event.source !== window.parent) {\n      return;\n    }\n    \n    switch (event.data.type) {\n      case 'START_INSPECTION':\n        startDOMInspection();\n        break;\n      case 'STOP_INSPECTION':\n        stopDOMInspection();\n        break;\n      case 'CLEAR_HIGHLIGHT':\n        clearHighlight();\n        break;\n      default:\n        break;\n    }\n  });\n  \n  // 暴露全局API（用于浏览器直接引入）\n  window.DOMInspector = {\n    start: startDOMInspection,\n    stop: stopDOMInspection,\n    clear: clearHighlight,\n    highlight: highlightElement,\n    getElementInfo: getElementInfo,\n    isActive: () => isInspecting\n  };\n\n})();\n";
// 生成DOM审查器脚本标签（JSX格式）
var generateDOMInspectorScript = function() {
    return '\n        <script \n          data-dom-inspector="true"\n          dangerouslySetInnerHTML={{\n            __html: `'.concat(DOM_INSPECTOR_SCRIPT, "`\n          }}\n        />");
};
// 注入DOM审查器脚本的通用函数
var injectDOMInspectorScriptToHTML = function(html) {
    return html.replace("</head>", "<script>".concat(DOM_INSPECTOR_SCRIPT, "</script></head>"));
};
// Vite插件（默认导出）
var viteAdapter = function() {
    return {
        name: "dom-inspector-injector",
        apply: "serve",
        transformIndexHtml: {
            order: "pre",
            handler: function handler(html) {
                return injectDOMInspectorScriptToHTML(html);
            }
        }
    };
};
// Ice框架适配器
var iceAdapter = function() {
    return {
        name: "ice-dom-inspector-injector",
        setup: function(param) {
            var onGetConfig = param.onGetConfig;
            // 使用onGetConfig来配置构建时的HTML转换
            onGetConfig(function(config) {
                if (config.mode !== "production") {
                    config.transformPlugins = config.transformPlugins || [];
                    config.transformPlugins.push({
                        name: "dom-inspector-transform",
                        transform: function(code, id) {
                            // 处理document.tsx文件
                            if (id.endsWith("document.tsx") || id.includes("document.tsx")) {
                                // 检查是否已经注入过脚本
                                if (code.includes("data-dom-inspector")) {
                                    return code;
                                }
                                // 使用模板生成DOM审查器脚本
                                var domInspectorScript = generateDOMInspectorScript();
                                // 在document.tsx中查找head标签并注入脚本
                                var transformedCode = code;
                                // 查找head标签的位置
                                var headMatch = code.match(/<head[^>]*>/i);
                                if (headMatch) {
                                    // 在head标签后注入脚本（保持JSX缩进）
                                    transformedCode = code.replace(headMatch[0], "".concat(headMatch[0], "\n        ").concat(domInspectorScript));
                                    return transformedCode;
                                }
                                return code;
                            }
                        }
                    });
                }
            });
        }
    };
};
// 手动注入函数
var injectDOMInspectorScript = function() {
    if (typeof document !== "undefined") {
        var scriptElement = document.createElement("script");
        scriptElement.innerHTML = DOM_INSPECTOR_SCRIPT;
        document.head.appendChild(scriptElement);
    }
};
// DOM审查器控制API
var DOMInspectorAPI = {
    // 开始审查模式
    startInspection: function() {
        if (typeof window !== "undefined") {
            window.postMessage({
                type: "START_INSPECTION"
            }, "*");
        }
    },
    // 停止审查模式
    stopInspection: function() {
        if (typeof window !== "undefined") {
            window.postMessage({
                type: "STOP_INSPECTION"
            }, "*");
        }
    },
    // 清除高亮
    clearHighlight: function() {
        if (typeof window !== "undefined") {
            window.postMessage({
                type: "CLEAR_HIGHLIGHT"
            }, "*");
        }
    },
    // 监听元素选择事件
    onElementSelected: function(callback) {
        if (typeof window !== "undefined") {
            var handleMessage = function(event) {
                if (event.data.type === "ELEMENT_SELECTED") {
                    callback(event.data.data);
                }
            };
            window.addEventListener("message", handleMessage);
            // 返回清理函数
            return function() {
                window.removeEventListener("message", handleMessage);
            };
        }
        return function() {}; // 返回空函数作为默认清理函数
    }
};
// 浏览器版本的DOM审查器类
var DOMInspectorInjector = /*#__PURE__*/ function() {
    "use strict";
    function DOMInspectorInjector() {
        _class_call_check._(this, DOMInspectorInjector);
        _define_property._(this, "scriptInjected", false);
        this.inject();
    }
    _create_class._(DOMInspectorInjector, [
        {
            // 注入脚本
            key: "inject",
            value: function inject() {
                if (this.scriptInjected) return;
                if (typeof document !== "undefined") {
                    var scriptElement = document.createElement("script");
                    scriptElement.innerHTML = DOM_INSPECTOR_SCRIPT;
                    document.head.appendChild(scriptElement);
                    this.scriptInjected = true;
                }
            }
        },
        {
            // 开始审查
            key: "start",
            value: function start() {
                if (typeof window !== "undefined" && window.DOMInspector) {
                    window.DOMInspector.start();
                }
            }
        },
        {
            // 停止审查
            key: "stop",
            value: function stop() {
                if (typeof window !== "undefined" && window.DOMInspector) {
                    window.DOMInspector.stop();
                }
            }
        },
        {
            // 清除高亮
            key: "clear",
            value: function clear() {
                if (typeof window !== "undefined" && window.DOMInspector) {
                    window.DOMInspector.clear();
                }
            }
        },
        {
            // 检查是否激活
            key: "isActive",
            value: function isActive() {
                if (typeof window !== "undefined" && window.DOMInspector) {
                    return window.DOMInspector.isActive();
                }
                return false;
            }
        },
        {
            // 监听元素选择事件
            key: "onElementSelected",
            value: function onElementSelected(callback) {
                if (typeof window !== "undefined") {
                    var handleEvent = function(event) {
                        callback(event.detail);
                    };
                    window.addEventListener("dom-inspector:element-selected", handleEvent);
                    // 返回清理函数
                    return function() {
                        window.removeEventListener("dom-inspector:element-selected", handleEvent);
                    };
                }
                return function() {};
            }
        }
    ]);
    return DOMInspectorInjector;
}();
// 浏览器环境下的自动注入和全局变量暴露
if (typeof window !== "undefined") {
    // 创建全局实例
    window.DOMInspectorInjector = DOMInspectorInjector;
    // 自动注入
    var inspector = new DOMInspectorInjector();
    window.domInspector = inspector;
}
var _default = {
    viteAdapter: viteAdapter,
    iceAdapter: iceAdapter,
    injectDOMInspectorScriptToHTML: injectDOMInspectorScriptToHTML,
    injectDOMInspectorScript: injectDOMInspectorScript,
    DOMInspectorAPI: DOMInspectorAPI,
    DOMInspectorInjector: DOMInspectorInjector,
    DOM_INSPECTOR_SCRIPT: DOM_INSPECTOR_SCRIPT
};

// 标准 CommonJS 导出，确保 ESM 加载器能正确识别命名导出
module.exports = {
    viteAdapter: viteAdapter,
    iceAdapter: iceAdapter,
    injectDOMInspectorScriptToHTML: injectDOMInspectorScriptToHTML,
    injectScript: injectDOMInspectorScript,
    DOMInspectorAPI: DOMInspectorAPI,
    DOMInspectorInjector: DOMInspectorInjector,
    DOM_INSPECTOR_SCRIPT: DOM_INSPECTOR_SCRIPT,
    default: _default
};
