// @ali/aone-super-error-capture/esm/index.js
// 错误捕获脚本模板
var ERROR_CAPTURE_SCRIPT = "\n  (function() {\n    // 捕获全局错误（包括资源加载错误）\n    window.addEventListener('error', function(event) {\n      if (window.parent && window.parent !== window) {\n        const errorData = {\n          type: event.type,\n          message: event.message,\n          filename: event.filename,\n          lineno: event.lineno,\n          colno: event.colno,\n          timeStamp: event.timeStamp\n        };\n        \n        // Error 对象（JavaScript 运行时错误）\n        if (event.error) {\n          errorData.error = {\n            message: event.error.message,\n            stack: event.error.stack,\n            name: event.error.name\n          };\n        }\n        \n        // Target 信息（资源加载错误）\n        if (event.target && event.target !== window) {\n          errorData.target = {\n            tagName: event.target.tagName,\n            src: event.target.src,\n            href: event.target.href\n          };\n        }\n        \n        window.parent.postMessage({\n          type: 'CONSOLE_ERROR',\n          error: errorData,\n          timestamp: new Date().toISOString()\n        }, '*');\n      }\n    }, true);\n\n    // 捕获Promise错误\n    window.addEventListener('unhandledrejection', function(event) {\n      if (window.parent && window.parent !== window) {\n        const errorData = {\n          type: event.type,\n          timeStamp: event.timeStamp\n        };\n        \n        // 序列化 reason\n        if (event.reason) {\n          if (typeof event.reason === 'object' && event.reason !== null) {\n            errorData.reason = {\n              message: event.reason.message,\n              stack: event.reason.stack,\n              name: event.reason.name\n            };\n          } else {\n            errorData.reason = String(event.reason);\n          }\n        }\n        \n        window.parent.postMessage({\n          type: 'CONSOLE_ERROR',\n          error: errorData,\n          timestamp: new Date().toISOString()\n        }, '*');\n      }\n    });\n  })();\n";
// 生成错误捕获脚本标签（JSX格式）
var generateErrorCaptureScript = function() {
    return '\n        <script \n          data-error-capture="true"\n          dangerouslySetInnerHTML={{\n            __html: `'.concat(ERROR_CAPTURE_SCRIPT, "`\n          }}\n        />");
};
// 注入错误捕获脚本的通用函数
function injectErrorScript(html) {
    return html.replace("</head>", "<script>".concat(ERROR_CAPTURE_SCRIPT, "</script></head>"));
}
// Vite插件（默认导出）
var viteAdapter = function() {
    return {
        name: "error-capture",
        apply: "serve",
        transformIndexHtml: {
            order: "pre",
            handler: function handler(html) {
                return injectErrorScript(html);
            }
        }
    };
};
// Ice框架适配器
var iceAdapter = function() {
    return {
        name: "ice-error-capture",
        setup: function(param) {
            var onGetConfig = param.onGetConfig;
            console.log("\uD83D\uDE80 Error capture plugin: Plugin setup started");
            // 使用onGetConfig来配置构建时的HTML转换
            onGetConfig(function(config) {
                if (config.mode !== "production") {
                    // 使用transformPlugins配置
                    config.transformPlugins = config.transformPlugins || [];
                    config.transformPlugins.push({
                        name: "error-capture-transform",
                        transform: function(code, id) {
                            // 处理document.tsx文件
                            if (id.endsWith("document.tsx") || id.includes("document.tsx")) {
                                // 检查是否已经注入过脚本
                                if (code.includes("data-error-capture")) {
                                    return code;
                                }
                                // 使用模板生成错误捕获脚本
                                var errorScript = generateErrorCaptureScript();
                                // 在document.tsx中查找head标签并注入脚本
                                var transformedCode = code;
                                // 查找head标签的位置
                                var headMatch = code.match(/<head[^>]*>/i);
                                if (headMatch) {
                                    // 在head标签后注入脚本（保持JSX缩进）
                                    transformedCode = code.replace(headMatch[0], "".concat(headMatch[0], "\n        ").concat(errorScript));
                                }
                                return transformedCode;
                            }
                            return code;
                        }
                    });
                }
            });
        }
    };
};
// 手动注入函数
var injectErrorCaptureScript = function() {
    if (typeof document !== "undefined") {
        var scriptElement = document.createElement("script");
        scriptElement.innerHTML = ERROR_CAPTURE_SCRIPT;
        document.head.appendChild(scriptElement);
    }
};
// 浏览器环境下的自动注入
if (typeof window !== "undefined") {
    // 自动注入错误捕获脚本
    injectErrorCaptureScript();
}
// 默认导出（Vite插件）
export default {
    viteAdapter: viteAdapter,
    iceAdapter: iceAdapter,
    injectErrorCaptureScript: injectErrorCaptureScript,
    injectErrorScript: injectErrorScript
};
// 命名导出所有适配器和工具
export { viteAdapter, iceAdapter, injectErrorCaptureScript, injectErrorScript };
